src/main/java/ro/msg/learning/shop/services/TestServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package ro.msg.learning.shop.services;
+
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.services.interfaces.TestService;
+import ro.msg.learning.shop.entitites.*;
+import ro.msg.learning.shop.repositories.*;
+import ro.msg.learning.shop.utils.TestingUtils;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class TestServiceImpl implements TestService {
+	@Autowired
+	private final ProductRepository productRepository;
+	@Autowired
+	private final LocationRepository locationRepository;
+	@Autowired
+	private final ProductCategoryRepository productCategoryRepository;
+	@Autowired
+	private final SupplierRepository supplierRepository;
+	@Autowired
+	private final CustomerRepository customerRepository;
+	@Autowired
+	private final StockRepository stockRepository;
+	@Autowired
+	private final OrderRepository orderRepository;
+	@Autowired
+	private final OrderDetailRepository orderDetailRepository;
+
+	@Override
+	@Transactional
+	public void populateDatabase() {
+
+		List<EntityWithID> entities = TestingUtils.getEntities();
+		entities.forEach(entity -> saveEntity(entity));
+	}
+
+	@Override
+	@Transactional
+	public void clearDatabase() {
+			orderDetailRepository.deleteAllInBatch();
+			orderRepository.deleteAllInBatch();
+			customerRepository.deleteAllInBatch();
+			stockRepository.deleteAllInBatch();
+			locationRepository.deleteAllInBatch();
+			productRepository.deleteAllInBatch();
+			productCategoryRepository.deleteAllInBatch();
+			supplierRepository.deleteAllInBatch();
+	}
+
+	private void saveEntity(EntityWithID entity) {
```

Comment:
Please do a refactoring here, this method is very unmaintainable. You can create it using generics and adding also a parameter for repository with the type as JpaRepository<T, UUID> where T is the generic type.

Reply:
Refactored.